aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -190,6 +215,46 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    fun ed25519_verify_t<T: drop> (signature: vector<u8>, public_key: vector<u8>, data: T): bool {
+        let encoded = Proof {
+            type_info: type_info::type_of<T>(),
+            inner: data,
+        };
+        signature::ed25519_verify(signature, public_key, bcs::to_bytes(&encoded))
+    }
+
+    public entry fun rotate_authentication_key_ed25519(account: &signer, new_public_key: vector<u8>, signature: vector<u8>) acquires Account, OriginatingAddress {
```

Comment:
Are we still keeping the other rotate auth key?

Reply:
yes - we want to give the user an option to opt out of this function so we're also keeping the old one

Reply:
Hmmm I found it confusing to have both. If we think this is immensely useful despite lack of parallelism (multiple txs calling this function cannot be executed in parallel as they all modify the same resource), we should only keep this IMO.

Reply:
I think this one will cost more gas as we're reading and writing into a global map, so we're letting the developers choose which one to call. I get that it could be confusing to have both tho

Reply:
Okay